File: ProblemSet/509.fibonacci-number/509.fibonacci-number 公式法.java
/*
 * @lc app=leetcode id=509 lang=java
 *
 * [509] Fibonacci Number
 */

// @lc code=start
class Solution {
    public int fib(int N) {
        double glodenRatio = (1 + Math.sqrt(5)) / 2;
        return (int)Math.round(Math.pow(glodenRatio, N) / Math.sqrt(5));
    }
}
// @lc code=end

File: ProblemSet/509.fibonacci-number/python/509.fibonacci-number 公式法.py
#
# @lc app=leetcode id=509 lang=python3
#
# [509] Fibonacci Number
#

# @lc code=start
class Solution:
    def fib(self, n: int) -> int:
        gloden_ratio = (1 + 5 ** 0.5) / 2
        return round(gloden_ratio ** n / (5 ** 0.5))
        
        
# @lc code=end

